dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -60,7 +60,28 @@ def new_daily_foorm
     #
     # If the pre-survey has been already completed, will redirect to thanks page.
     def new_pre_foorm
-      new_general_foorm(survey_names: PRE_SURVEY_CONFIG_PATHS, day: 0)
```

Comment:
I didn't want to modify `new_general_foorm` since that's used for day surveys and post-workshop surveys so I tried to move all the relevant logic from the methods that `new_general_foorm` calls into `new_pre_foorm` so that it can be handled separately since it now works differently.